ocw/lib/gce.py

```diff
@@ -207,3 +215,129 @@ def cleanup_all(self):
+            'validBeforeTime': '2021-12-02T12:13:43Z',
+            'keyAlgorithm': 'KEY_ALG_RSA_2048',
+            'keyOrigin': 'GOOGLE_PROVIDED',
+            'keyType': 'USER_MANAGED'}
+         ]}
+
+        Using `validAfterTime` the function calculate which Service Accounts
+        should be deleted. The actual delete function takes the email in its
+        request, so the `filter_email_service_accounts` has to returned them
+        corelated email addresses of the `service_resp['keys']`
+
+        Parameters
+        ----------
+        service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        vault_account_tuple : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        time_limit : int
+            the number of hours where the Service Accounts are keep been
+            ignored from the cleanup job
+        '''
+        from datetime import datetime, timedelta
+        time_limit = datetime.now() - timedelta(hours=time_limit)
+        dt_frm = datetime.fromisoformat
+        filtered_service_accounts = []
+        for email in vault_account_tuple:
+            service_resp = service.projects().serviceAccounts().keys().list(
+                name='projects/-/serviceAccounts/%s' % email).execute()
+            filtered_service_accounts += [k['name'] for k in service_resp['keys'] if dt_frm(
+                    k['validAfterTime'][:-1]) < time_limit]
+        return list(filter(lambda e: e in str(filtered_service_accounts), vault_account_tuple))
+
+    @filterService(name='vaultopenqa')
+    def parser_service_account_list(self, iam_service):
+        ''' Returns a list of the emails of the Service Accounts
+
+        The request is using the resource name of the project associated with
+        the service accounts. This returns a list of dict items, which are
+        also not vault related.
+        The objects returned in chunks which contain a `nextPageToken`
+        to the next page. The `parser_service_account_list` parses each page
+        and returns the full list. Subsequently, the list is filtered to return
+        only emails correlated with vault Service Accounts
+
+        Parameters
+        ----------
+        iam_service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        '''
+        service_accounts_emails = []
+        req = iam_service.projects().serviceAccounts().list(
+            name='projects/suse-sle-qa')
+
+        while True:
+            resp = req.execute()
+            service_accounts_emails += [service_account['email']
+                                        for service_account in resp.get('accounts', [])]
+
+            req = iam_service.projects().serviceAccounts().list_next(
+                previous_request=req, previous_response=resp)
+            if req is None:
+                break
```

Comment:
since very next line after this break is return anyway we can simply do this 
```suggestion
                return service_accounts_emails
```

Reply:
*this change is ready for review*